sdk/storage/storage-queue/test/utils/recorder.ts

```diff
@@ -0,0 +1,326 @@
+import { blobToString } from "./index.browser";
+import * as dotenv from "dotenv";
+dotenv.config({ path: "../../.env" });
+
+let nock: any;
+if (!isBrowser()) {
+  nock = require("nock");
+}
+
+/**
+ * Possible reasons for skipping a test:
+ * * Abort: browser testing unexpectedly finishes when a request is aborted during playback (unknown reason; probably related to the way nise handles it)
+ * * Character: there are characters in the message that are not supported in browser logging or in ECMAScript
+ * * Progress: Nock does not record a request if it's aborted in a 'progress' callback
+ * * Size: the generated recording file is too big and would considerably increase the size of the package
+ * * Tempfile: the request makes use of a random tempfile created locally, and the recorder does not support recording it as unique information
+ * * UUID: a UUID is randomly generated within the SDK and used in an HTTP request, resulting in Nock being unable to recognize it
+ */
+const skip = [
+  // Abort
+  "browsers/aborter/recording_should_abort_after_aborter_timeout.json",
+  // Abort
+  "browsers/aborter/recording_should_abort_after_parent_aborter_calls_abort.json",
+  // Abort
+  "browsers/aborter/recording_should_abort_after_parent_aborter_timeout.json",
+  // Abort
+  "browsers/aborter/recording_should_abort_when_calling_abort_before_request_finishes.json",
+  // Character
+  "browsers/messagesurl/recording_enqueue_peek_dequeue_special_characters.json"
+];
+
+abstract class Recorder {
+  protected readonly filepath: string;
+  public uniqueTestInfo: any = {};
+
+  constructor(env: string, testHierarchy: string, testTitle: string, ext: string) {
+    this.filepath =
+      env +
+      "/" +
+      this.formatPath(testHierarchy) +
+      "/recording_" +
+      this.formatPath(testTitle) +
+      "." +
+      ext;
+  }
+
+  protected formatPath(path: string): string {
+    return path
+      .toLowerCase()
+      .replace(/ /g, "_")
+      .replace(/<=/g, "lte")
+      .replace(/>=/g, "gte")
+      .replace(/</g, "lt")
+      .replace(/>/g, "gt")
+      .replace(/=/g, "eq")
+      .replace(/\W/g, "");
+  }
+
+  public skip(): boolean {
+    return skip.includes(this.filepath);
+  }
+
+  public abstract record(): void;
+  public abstract playback(): void;
+  public abstract stop(): void;
+}
+
+class NockRecorder extends Recorder {
+  constructor(testHierarchy: string, testTitle: string) {
+    super("node", testHierarchy, testTitle, "js");
+  }
+
+  public record(): void {
+    nock.recorder.rec({
+      dont_print: true
+    });
+  }
+
+  public playback(): void {
+    this.uniqueTestInfo = require("../recordings/" + this.filepath).testInfo;
+  }
+
+  public stop(): void {
+    const importNock = "let nock = require('nock');\n";
+    const fixtures = nock.recorder.play();
+
+    // It's important to print writing errors because some tests end up catching them
+    const file = fs.createWriteStream("./recordings/" + this.filepath, { flags: "w" });
```

Comment:
@kinelski 
With the current way of doing things, we need to manually create the directories with the describe-block titles in order to allow recording to happen.
I think we should consider automating this step.

Reply:
https://github.com/HarshaNalluru/azure-sdk-for-js/compare/RecordAndPlaybackNock...HarshaNalluru:RecordAndPlaybackCreateFolder?expand=1